optimum/intel/openvino/modeling.py

```diff
@@ -464,17 +461,33 @@ def _from_transformers(
             force_download=force_download,
             trust_remote_code=trust_remote_code,
             ov_config=ov_config,
-            library_name="transformers",
+            library_name=library_name,
         )
 
         config.save_pretrained(save_dir_path)
-        return cls._from_pretrained(
+        model = cls._from_pretrained(
             model_id=save_dir_path,
             config=config,
             load_in_8bit=load_in_8bit,
             quantization_config=quantization_config,
             **kwargs,
         )
+        if len(model.model.outputs) == 2:  # Sentence Transormers output
+            from .modeling_sentence_transformers import OVModelForSTFeatureExtraction
+
+            model = OVModelForSTFeatureExtraction(
+                model=model.model,
+                config=config,
+                model_save_dir=save_dir_path,
+                quantization_config=quantization_config,
+                model_id=model_id,
+                **kwargs,
+            )
+            return model
+        else:
+            if is_sentence_transformers_available():
```

Comment:
I do not think that this warning is relevant as it is expected to get transformers-based model instance (from user understanding it is desire behaviour if this model class selected). I think it is better provide opposite warning if we understand that sentence-transformers loaded in this class and recommend to use dedicated class explicitly